api/internal/handler/server_info/server_info_test.go

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package server_info
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/shiningrush/droplet"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/apisix/manager-api/internal/core/entity"
+	"github.com/apisix/manager-api/internal/core/storage"
+	"github.com/apisix/manager-api/internal/core/store"
+)
+
+const (
+	SleepTime = 100 * time.Millisecond
+)
+
+var serverInfoHandler *Handler
+
+func init() {
+	err := storage.InitETCDClient([]string{"127.0.0.1:2379"})
+
+	if err != nil {
+		panic(err)
+	}
+	err = store.InitStores()
+	if err != nil {
+		panic(err)
+	}
+
+	serverInfoHandler = &Handler{
+		serverInfoStore: store.GetStore(store.HubKeyServerInfo),
+	}
+}
+
+func create(t *testing.T, ctx droplet.Context, info *entity.ServerInfo) {
```

Comment:
Shouldn't we use a mocked store here? Rather than depending on a real etcd cluster.

Reply:
Many unit tests depend on a real etcd cluster.
I think maybe we need to create a TODO to refactor those test cases.

Reply:
@starsz  we have started to refactor the unit test cases. you can refer : https://github.com/apache/apisix-dashboard/blob/master/api/internal/handler/consumer/consumer_test.go

Reply:
Got it.